activerecord/lib/active_record/connection_adapters/pool_config.rb

```diff
@@ -41,11 +40,33 @@ def server_version(connection)
         @server_version || synchronize { @server_version ||= connection.get_database_version }
       end
 
+      def connection_class=(connection_class)
+        case connection_class
+        when ConnectionHandler::StringConnectionName
+          @connection_class_name = connection_class
+        else
+          @connection_class_name = connection_class.name
```

Comment:
Can we ever get an anonymous class here?

Reply:
I suppose the answer is "yes", since I can do this:

```
$ bin/rails c
Loading development environment (Rails 8.0.1)
>> k = Class.new(ActiveRecord::Base) { self.table_name = "posts" }
=> #<Class:0x00007dc40fed1ba0> (call '#<Class:0x00007dc40fed1ba0>.load_schema' to load schema informations)
>> k.first
   Load (0.1ms)  SELECT "posts".* FROM "posts" ORDER BY "posts"."id" ASC LIMIT 1 /*application='Test'*/
=> nil
>> k.name
=> nil
```

I don't know how common this is in the real world. But since the answer is "yes" this PR is insufficient as currently written.